### PR TITLE
RFC: Noise Protocol

### PR DESCRIPTION
**Pull request by kim**
[Rendered](https://github.com/radicle-dev/radicle-link/blob/rfc/noise/docs/rfc/0000-noise.md)

### CODE REVIEW

File: docs/rfc/0000-noise.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+[Noise][noise] is preferrable over [TLS][tls] in the context of peer-to-peer
+networking, because it is considerably simpler and more compact.  Noise also
+doesn't mandate implementation of a PKI, which, in an open peer-to-peer network,
+is not applicable. Deliberately defeating the PKI mechanisms of TLS for our
+purposes, however, yields dubious security properties, and may be a source of
+bugs in implementations.
+
+Recently, the [ipfs-embed] project has released an experimental implementation
+of [Noise transport security for QUIC][quinn-noise], and registered a [QUIC
+version range][QUIC-version]. Due to the similarities in purpose and
```
Comment (FintanH) [inline docs/rfc/0000-noise.md:28]:
```suggestion
version range][QUIC-versions]. Due to the similarities in purpose and
```

File: docs/rfc/0000-noise.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+transport-layer security, and is still in an experimental stage. It is thus
+debatable if auditing is a prerequisite for adopting the implementation. On the
+other hand, `ipfs-embed` is developed in the context of the larger IPFS
+ecosystem, and itself in a very early stage. It is likely that the
+implementation will receive more scrutiny as it matures.
+
+
+### Key Reuse
+
+For completeness, we point out that Noise explicitly discourages re-use of
```
Comment (FintanH) [inline docs/rfc/0000-noise.md:112]: Do you have any pointers to where this is discouraged?

**Review by FintanH — COMMENTED** [review]
The recommendation makes sense :+1: 

Few editorial things. There are some links missing:
* ALPN
* snow
* xoodyak
* xoodoo
* rustls

**Review by xla — APPROVED** [review]
🅿️ 🎎 📫 🛣

**Review by FintanH — DISMISSED** [review]

**Pull request merged**